// @react-spectrum/combobox/dist/module.js
import { useProviderProps, useProvider } from "@react-spectrum/provider";
import { useLabel } from "@react-aria/label";
import { useFilter, useMessageFormatter } from "@react-aria/i18n";
import { useDialog } from "@react-aria/dialog";
import { useComboBox } from "@react-aria/combobox";
import { useButton } from "@react-aria/button";
import { Tray, Popover } from "@react-spectrum/overlays";
import { TextFieldBase } from "@react-spectrum/textfield";
import { setInteractionModality, useHover, PressResponder } from "@react-aria/interactions";
import _react, { useCallback, useEffect, useRef, useState } from "react";
import { ProgressCircle } from "@react-spectrum/progress";
import { mergeProps, useId, useLayoutEffect } from "@react-aria/utils";
import { useComboBoxState } from "@react-stately/combobox";
import _spectrumIconsUiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import _spectrumIconsUiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import { ListBoxBase, useListBoxLayout } from "@react-spectrum/listbox";
import { FocusRing, FocusScope, focusSafely } from "@react-aria/focus";
import { FieldButton, ClearButton } from "@react-spectrum/button";
import { Field } from "@react-spectrum/label";
import { DismissButton, useOverlayPosition, useOverlayTrigger } from "@react-aria/overlays";
import { classNames, useFocusableRef, useIsMobileDevice, useResizeObserver, useUnwrapDOMRef, unwrapDOMRef } from "@react-spectrum/utils";
import _spectrumIconsUiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import _babelRuntimeHelpersEsmInteropRequireDefault from "@babel/runtime/helpers/esm/interopRequireDefault";
import _babelRuntimeHelpersEsmExtends from "@babel/runtime/helpers/esm/extends";
import { Item, Section } from "@react-stately/collections";
export { Item, Section };
import "./main.css";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/src/combobox.css
var $dea09d50cdec57db7ad515930d6daf$exports = {};
$dea09d50cdec57db7ad515930d6daf$exports = {
  "no-results": "_no-results_b1b77",
  "mobile-combobox": "_mobile-combobox_b1b77",
  "mobile-input": "_mobile-input_b1b77",
  "mobile-value": "_mobile-value_b1b77",
  "tray-dialog": "_tray-dialog_b1b77",
  "tray-textfield": "_tray-textfield_b1b77",
  "has-label": "_has-label_b1b77",
  "tray-textfield-input": "_tray-textfield-input_b1b77",
  "tray-listbox": "_tray-listbox_b1b77"
};
// ASSET: /Users/govett/dev/react-spectrum-v3/node_modules/@adobe/spectrum-css-temp/components/button/vars.css
var $a0b8eb4b46824fd288195a4db8814b$exports = {};
$a0b8eb4b46824fd288195a4db8814b$exports = {
  "spectrum-Button": "_spectrum-Button_7a745",
  "spectrum-ActionButton": "_spectrum-ActionButton_7a745",
  "spectrum-LogicButton": "_spectrum-LogicButton_7a745",
  "spectrum-FieldButton": "_spectrum-FieldButton_7a745",
  "spectrum-ClearButton": "_spectrum-ClearButton_7a745",
  "focus-ring": "_focus-ring_7a745",
  "spectrum-Icon": "_spectrum-Icon_7a745",
  "spectrum-ActionButton--emphasized": "_spectrum-ActionButton--emphasized_7a745",
  "spectrum-ActionButton--staticColor": "_spectrum-ActionButton--staticColor_7a745",
  "is-hovered": "_is-hovered_7a745",
  "spectrum-Button-label": "_spectrum-Button-label_7a745",
  "spectrum-ActionButton-label": "_spectrum-ActionButton-label_7a745",
  "spectrum-ActionButton-hold": "_spectrum-ActionButton-hold_7a745",
  "spectrum-ActionButton--quiet": "_spectrum-ActionButton--quiet_7a745",
  "is-disabled": "_is-disabled_7a745",
  "is-open": "_is-open_7a745",
  "spectrum-FieldButton--quiet": "_spectrum-FieldButton--quiet_7a745",
  "spectrum-ClearButton--overBackground": "_spectrum-ClearButton--overBackground_7a745",
  "spectrum-ClearButton--small": "_spectrum-ClearButton--small_7a745",
  "is-focused": "_is-focused_7a745",
  "is-active": "_is-active_7a745",
  "spectrum-Button--cta": "_spectrum-Button--cta_7a745",
  "spectrum-Button--primary": "_spectrum-Button--primary_7a745",
  "spectrum-Button--secondary": "_spectrum-Button--secondary_7a745",
  "spectrum-Button--warning": "_spectrum-Button--warning_7a745",
  "spectrum-Button--overBackground": "_spectrum-Button--overBackground_7a745",
  "spectrum-Button--quiet": "_spectrum-Button--quiet_7a745",
  "is-selected": "_is-selected_7a745",
  "spectrum-ActionButton--staticWhite": "_spectrum-ActionButton--staticWhite_7a745",
  "spectrum-ActionButton--staticBlack": "_spectrum-ActionButton--staticBlack_7a745",
  "spectrum-LogicButton--and": "_spectrum-LogicButton--and_7a745",
  "spectrum-LogicButton--or": "_spectrum-LogicButton--or_7a745",
  "is-placeholder": "_is-placeholder_7a745",
  "spectrum-FieldButton--invalid": "_spectrum-FieldButton--invalid_7a745"
};
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/ar-AE.json
var $aeabefb3f029f59bb5fb8536b65cc107$exports = {};
$aeabefb3f029f59bb5fb8536b65cc107$exports = JSON.parse("{\"clear\":\"مسح\",\"invalid\":\"(غير صالح)\",\"loading\":\"جار التحميل...\",\"noResults\":\"لا توجد نتائج\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/bg-BG.json
var $e81215b32fa7db9a7878fa16113eb13d$exports = {};
$e81215b32fa7db9a7878fa16113eb13d$exports = JSON.parse("{\"clear\":\"Изчисти\",\"invalid\":\"(невалиден)\",\"loading\":\"Зареждане...\",\"noResults\":\"Няма резултати\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/cs-CZ.json
var $bb869dea6e09984e8b04fe440dc02ec$exports = {};
$bb869dea6e09984e8b04fe440dc02ec$exports = JSON.parse("{\"clear\":\"Vymazat\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítání...\",\"noResults\":\"Žádné výsledky\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/da-DK.json
var $eaf0e4fc84da4e97fcf8b24a0663f670$exports = {};
$eaf0e4fc84da4e97fcf8b24a0663f670$exports = JSON.parse("{\"clear\":\"Ryd\",\"invalid\":\"(ugyldig)\",\"loading\":\"Indlæser ...\",\"noResults\":\"Ingen resultater\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/de-DE.json
var $b56bf5d22cb0f8734a4fc5ee20a9$exports = {};
$b56bf5d22cb0f8734a4fc5ee20a9$exports = JSON.parse("{\"clear\":\"Löschen\",\"invalid\":\"(ungültig)\",\"loading\":\"Wird geladen...\",\"noResults\":\"Keine Ergebnisse\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/el-GR.json
var $e2505be9c44ad2de362aed02038721c6$exports = {};
$e2505be9c44ad2de362aed02038721c6$exports = JSON.parse("{\"clear\":\"Καθαρισμός\",\"invalid\":\"(δεν ισχύει)\",\"loading\":\"Φόρτωση...\",\"noResults\":\"Χωρίς αποτέλεσμα\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/en-US.json
var $d463e3a7667fca55cd060af86bae6dc7$exports = {};
$d463e3a7667fca55cd060af86bae6dc7$exports = JSON.parse("{\"loading\":\"Loading...\",\"noResults\":\"No results\",\"clear\":\"Clear\",\"invalid\":\"(invalid)\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/es-ES.json
var $d4e2ba5b2cf0711ecf714c846bf7985$exports = {};
$d4e2ba5b2cf0711ecf714c846bf7985$exports = JSON.parse("{\"clear\":\"Borrar\",\"invalid\":\"(no válido)\",\"loading\":\"Cargando...\",\"noResults\":\"Sin resultados\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/et-EE.json
var $e3596e15452c69f045c20524dbb2$exports = {};
$e3596e15452c69f045c20524dbb2$exports = JSON.parse("{\"clear\":\"Puhasta\",\"invalid\":\"(kehtetu)\",\"loading\":\"Laadimine...\",\"noResults\":\"Tulemusi pole\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/fi-FI.json
var $c945fdb8634b778ebd85cce64d9f$exports = {};
$c945fdb8634b778ebd85cce64d9f$exports = JSON.parse("{\"clear\":\"Kirkas\",\"invalid\":\"(epäkelpo)\",\"loading\":\"Ladataan...\",\"noResults\":\"Ei tuloksia\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/fr-FR.json
var $ddf38499897200fd5686ac2871649df$exports = {};
$ddf38499897200fd5686ac2871649df$exports = JSON.parse("{\"clear\":\"Effacer\",\"invalid\":\"(non valide)\",\"loading\":\"Chargement en cours...\",\"noResults\":\"Aucun résultat\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/he-IL.json
var $a41bddea8c9d7e8d301620d4e42f$exports = {};
$a41bddea8c9d7e8d301620d4e42f$exports = JSON.parse("{\"clear\":\"נקי\",\"invalid\":\"(לא חוקי)\",\"loading\":\"טוען...\",\"noResults\":\"אין תוצאות\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/hr-HR.json
var $c91221142608664f57adc7c440d25c5c$exports = {};
$c91221142608664f57adc7c440d25c5c$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/hu-HU.json
var $fbc3e6ba153b38828dbd4e51aedcd9c$exports = {};
$fbc3e6ba153b38828dbd4e51aedcd9c$exports = JSON.parse("{\"clear\":\"Törlés\",\"invalid\":\"(érvénytelen)\",\"loading\":\"Betöltés folyamatban…\",\"noResults\":\"Nincsenek találatok\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/it-IT.json
var $e136b067ceb9d5b9a035ed9f5faa6018$exports = {};
$e136b067ceb9d5b9a035ed9f5faa6018$exports = JSON.parse("{\"clear\":\"Cancella\",\"invalid\":\"(non valido)\",\"loading\":\"Caricamento in corso...\",\"noResults\":\"Nessun risultato\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/ja-JP.json
var $bc0662f1900351b0445924c5be7$exports = {};
$bc0662f1900351b0445924c5be7$exports = JSON.parse("{\"clear\":\"クリア\",\"invalid\":\"(無効)\",\"loading\":\"読み込み中...\",\"noResults\":\"結果なし\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/ko-KR.json
var $aae5c87ddbf07a13d7cf3cb3c706c096$exports = {};
$aae5c87ddbf07a13d7cf3cb3c706c096$exports = JSON.parse("{\"clear\":\"지우기\",\"invalid\":\"(유효하지 않음)\",\"loading\":\"로드 중...\",\"noResults\":\"결과 없음\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/lt-LT.json
var $f9bf28c16957dd98d784eef2160689$exports = {};
$f9bf28c16957dd98d784eef2160689$exports = JSON.parse("{\"clear\":\"Skaidrus\",\"invalid\":\"(netinkama)\",\"loading\":\"Įkeliama...\",\"noResults\":\"Be rezultatų\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/lv-LV.json
var $c41b94dd0457195a34a60e9d0893d89e$exports = {};
$c41b94dd0457195a34a60e9d0893d89e$exports = JSON.parse("{\"clear\":\"Notīrīt\",\"invalid\":\"(nederīgs)\",\"loading\":\"Notiek ielāde...\",\"noResults\":\"Nav rezultātu\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/nb-NO.json
var $f8ae30ecd9af5338c78bf67acc6fb392$exports = {};
$f8ae30ecd9af5338c78bf67acc6fb392$exports = JSON.parse("{\"clear\":\"Tøm\",\"invalid\":\"(ugyldig)\",\"loading\":\"Laster inn ...\",\"noResults\":\"Ingen resultater\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/nl-NL.json
var $f21b1c8f7ed4d2e3b8bb19291cb8$exports = {};
$f21b1c8f7ed4d2e3b8bb19291cb8$exports = JSON.parse("{\"clear\":\"Helder\",\"invalid\":\"(ongeldig)\",\"loading\":\"Laden...\",\"noResults\":\"Geen resultaten\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/pl-PL.json
var $e4cc01ee8904e31a9a335f923ae8629a$exports = {};
$e4cc01ee8904e31a9a335f923ae8629a$exports = JSON.parse("{\"clear\":\"Wyczyść\",\"invalid\":\"(nieprawidłowy)\",\"loading\":\"Trwa ładowanie...\",\"noResults\":\"Brak wyników\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/pt-BR.json
var $df8dca18e32a461efd140442fc548c72$exports = {};
$df8dca18e32a461efd140442fc548c72$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"Carregando...\",\"noResults\":\"Nenhum resultado\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/pt-PT.json
var $e5697b97bd7b08a87614a3a63c2233$exports = {};
$e5697b97bd7b08a87614a3a63c2233$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"A carregar...\",\"noResults\":\"Sem resultados\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/ro-RO.json
var $a4d94a6f59475598baa70097a6ac3345$exports = {};
$a4d94a6f59475598baa70097a6ac3345$exports = JSON.parse("{\"clear\":\"Golire\",\"invalid\":\"(nevalid)\",\"loading\":\"Se încarcă...\",\"noResults\":\"Niciun rezultat\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/ru-RU.json
var $ec693ed520530f049bb03a9de8f2$exports = {};
$ec693ed520530f049bb03a9de8f2$exports = JSON.parse("{\"clear\":\"Очистить\",\"invalid\":\"(недействительно)\",\"loading\":\"Загрузка...\",\"noResults\":\"Результаты отсутствуют\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/sk-SK.json
var $b620d822b146f7dbd3c5d9fd416f3640$exports = {};
$b620d822b146f7dbd3c5d9fd416f3640$exports = JSON.parse("{\"clear\":\"Vymazať\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítava sa...\",\"noResults\":\"Žiadne výsledky\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/sl-SI.json
var $f131bcd2c7e8f9c8460b2a7928760dc$exports = {};
$f131bcd2c7e8f9c8460b2a7928760dc$exports = JSON.parse("{\"clear\":\"Jasen\",\"invalid\":\"(neveljavno)\",\"loading\":\"Nalaganje...\",\"noResults\":\"Ni rezultatov\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/sr-SP.json
var $eeedef4ebbff571eaf41ecd136f71$exports = {};
$eeedef4ebbff571eaf41ecd136f71$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/sv-SE.json
var $db418128beb57024edf7ac9e84c764e1$exports = {};
$db418128beb57024edf7ac9e84c764e1$exports = JSON.parse("{\"clear\":\"Rensa\",\"invalid\":\"(ogiltigt)\",\"loading\":\"Läser in...\",\"noResults\":\"Inga resultat\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/tr-TR.json
var $eb9f1cce2dfb7c8c5b9ea893e144ce0d$exports = {};
$eb9f1cce2dfb7c8c5b9ea893e144ce0d$exports = JSON.parse("{\"clear\":\"Temizle\",\"invalid\":\"(geçersiz)\",\"loading\":\"Yükleniyor...\",\"noResults\":\"Sonuç yok\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/uk-UA.json
var $d7aab1629c77f8c4b88e2938949da4$exports = {};
$d7aab1629c77f8c4b88e2938949da4$exports = JSON.parse("{\"clear\":\"Очистити\",\"invalid\":\"(недійсне)\",\"loading\":\"Завантаження...\",\"noResults\":\"Результатів немає\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/zh-CN.json
var $f22a63a5549fed0b00db96652c08cf$exports = {};
$f22a63a5549fed0b00db96652c08cf$exports = JSON.parse("{\"clear\":\"透明\",\"invalid\":\"（无效）\",\"loading\":\"正在加载...\",\"noResults\":\"无结果\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-spectrum/combobox/intl/zh-TW.json
var $c5ea069e7a6e02a0a53f10506cd9b35$exports = {};
$c5ea069e7a6e02a0a53f10506cd9b35$exports = JSON.parse("{\"clear\":\"清除\",\"invalid\":\"(無效)\",\"loading\":\"正在載入...\",\"noResults\":\"無任何結果\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/node_modules/@adobe/spectrum-css-temp/components/fieldlabel/vars.css
var $c9fb1f4220ea2c59cb2e10f0e9828c6e$exports = {};
$c9fb1f4220ea2c59cb2e10f0e9828c6e$exports = {
  "spectrum-FieldLabel": "_spectrum-FieldLabel_36b9b",
  "spectrum-FieldLabel--positionSide": "_spectrum-FieldLabel--positionSide_36b9b",
  "spectrum-FieldLabel-requiredIcon": "_spectrum-FieldLabel-requiredIcon_36b9b",
  "spectrum-FieldLabel--alignEnd": "_spectrum-FieldLabel--alignEnd_36b9b",
  "spectrum-Field": "_spectrum-Field_36b9b",
  "spectrum-Field--positionTop": "_spectrum-Field--positionTop_36b9b",
  "spectrum-Field-field": "_spectrum-Field-field_36b9b",
  "spectrum-Field-field--multiline": "_spectrum-Field-field--multiline_36b9b",
  "spectrum-Field--positionSide": "_spectrum-Field--positionSide_36b9b",
  "spectrum-Form": "_spectrum-Form_36b9b",
  "spectrum-Form--positionSide": "_spectrum-Form--positionSide_36b9b",
  "spectrum-Form--positionTop": "_spectrum-Form--positionTop_36b9b",
  "spectrum-Form-itemLabel": "_spectrum-Form-itemLabel_36b9b",
  "is-disabled": "_is-disabled_36b9b"
};
// ASSET: /Users/govett/dev/react-spectrum-v3/node_modules/@adobe/spectrum-css-temp/components/search/vars.css
var $f7ba17202ccc7135cfe7128957b$exports = {};
$f7ba17202ccc7135cfe7128957b$exports = {
  "spectrum-Search": "_spectrum-Search_d08c7",
  "spectrum-ClearButton": "_spectrum-ClearButton_d08c7",
  "is-quiet": "_is-quiet_d08c7",
  "spectrum-Search-input": "_spectrum-Search-input_d08c7",
  "spectrum-Search--invalid": "_spectrum-Search--invalid_d08c7",
  "spectrum-Search--valid": "_spectrum-Search--valid_d08c7",
  "spectrum-Search--loadable": "_spectrum-Search--loadable_d08c7",
  "spectrum-Search-circleLoader": "_spectrum-Search-circleLoader_d08c7",
  "spectrum-Search-validationIcon": "_spectrum-Search-validationIcon_d08c7",
  "spectrum-Textfield": "_spectrum-Textfield_d08c7"
};
// ASSET: /Users/govett/dev/react-spectrum-v3/node_modules/@adobe/spectrum-css-temp/components/inputgroup/vars.css
var $ed163f34af6780259e0266609fb62099$exports = {};
$ed163f34af6780259e0266609fb62099$exports = {
  "spectrum-InputGroup": "_spectrum-InputGroup_d053e",
  "spectrum-FieldButton": "_spectrum-FieldButton_d053e",
  "is-disabled": "_is-disabled_d053e",
  "spectrum-InputGroup-field": "_spectrum-InputGroup-field_d053e",
  "spectrum-InputGroup-input-validationIcon": "_spectrum-InputGroup-input-validationIcon_d053e",
  "spectrum-InputGroup-input-circleLoader": "_spectrum-InputGroup-input-circleLoader_d053e",
  "spectrum-InputGroup-input": "_spectrum-InputGroup-input_d053e",
  "spectrum-InputGroup--quiet": "_spectrum-InputGroup--quiet_d053e",
  "spectrum-InputGroup-icon": "_spectrum-InputGroup-icon_d053e",
  "spectrum-InputGroup-popover--quiet": "_spectrum-InputGroup-popover--quiet_d053e",
  "spectrum-Datepicker--range": "_spectrum-Datepicker--range_d053e",
  "spectrum-Datepicker--datetimeRange": "_spectrum-Datepicker--datetimeRange_d053e",
  "spectrum-Datepicker-startField": "_spectrum-Datepicker-startField_d053e",
  "spectrum-Datepicker-endField": "_spectrum-Datepicker-endField_d053e",
  "spectrum-Datepicker--rangeDash": "_spectrum-Datepicker--rangeDash_d053e",
  "is-focused": "_is-focused_d053e",
  "spectrum-Datepicker-focusRing": "_spectrum-Datepicker-focusRing_d053e",
  "spectrum-InputGroup--invalid": "_spectrum-InputGroup--invalid_d053e",
  "is-hovered": "_is-hovered_d053e",
  "focus-ring": "_focus-ring_d053e",
  "is-selected": "_is-selected_d053e",
  "spectrum-FieldButton--invalid": "_spectrum-FieldButton--invalid_d053e"
};
// ASSET: /Users/govett/dev/react-spectrum-v3/node_modules/@adobe/spectrum-css-temp/components/textfield/vars.css
var $aa1c742e7a2a16aa7981ca9e2f4a89$exports = {};
$aa1c742e7a2a16aa7981ca9e2f4a89$exports = {
  "spectrum-Textfield": "_spectrum-Textfield_61339",
  "spectrum-Textfield--quiet": "_spectrum-Textfield--quiet_61339",
  "spectrum-Textfield--multiline": "_spectrum-Textfield--multiline_61339",
  "spectrum-Textfield-input": "_spectrum-Textfield-input_61339",
  "is-placeholder": "_is-placeholder_61339",
  "is-hovered": "_is-hovered_61339",
  "spectrum-Textfield--valid": "_spectrum-Textfield--valid_61339",
  "spectrum-Textfield--invalid": "_spectrum-Textfield--invalid_61339",
  "spectrum-Textfield--loadable": "_spectrum-Textfield--loadable_61339",
  "spectrum-Textfield-validationIcon": "_spectrum-Textfield-validationIcon_61339",
  "spectrum-Textfield-icon": "_spectrum-Textfield-icon_61339",
  "spectrum-Textfield-inputIcon": "_spectrum-Textfield-inputIcon_61339",
  "spectrum-Textfield-circleLoader": "_spectrum-Textfield-circleLoader_61339",
  "is-focused": "_is-focused_61339",
  "focus-ring": "_focus-ring_61339",
  "is-disabled": "_is-disabled_61339"
};
// @ts-ignore
const $f30f10b3f7806340277c0b2046944ca$var$intlMessages = {
  "ar-AE": _babelRuntimeHelpersEsmInteropRequireDefault($aeabefb3f029f59bb5fb8536b65cc107$exports).default,
  "bg-BG": _babelRuntimeHelpersEsmInteropRequireDefault($e81215b32fa7db9a7878fa16113eb13d$exports).default,
  "cs-CZ": _babelRuntimeHelpersEsmInteropRequireDefault($bb869dea6e09984e8b04fe440dc02ec$exports).default,
  "da-DK": _babelRuntimeHelpersEsmInteropRequireDefault($eaf0e4fc84da4e97fcf8b24a0663f670$exports).default,
  "de-DE": _babelRuntimeHelpersEsmInteropRequireDefault($b56bf5d22cb0f8734a4fc5ee20a9$exports).default,
  "el-GR": _babelRuntimeHelpersEsmInteropRequireDefault($e2505be9c44ad2de362aed02038721c6$exports).default,
  "en-US": _babelRuntimeHelpersEsmInteropRequireDefault($d463e3a7667fca55cd060af86bae6dc7$exports).default,
  "es-ES": _babelRuntimeHelpersEsmInteropRequireDefault($d4e2ba5b2cf0711ecf714c846bf7985$exports).default,
  "et-EE": _babelRuntimeHelpersEsmInteropRequireDefault($e3596e15452c69f045c20524dbb2$exports).default,
  "fi-FI": _babelRuntimeHelpersEsmInteropRequireDefault($c945fdb8634b778ebd85cce64d9f$exports).default,
  "fr-FR": _babelRuntimeHelpersEsmInteropRequireDefault($ddf38499897200fd5686ac2871649df$exports).default,
  "he-IL": _babelRuntimeHelpersEsmInteropRequireDefault($a41bddea8c9d7e8d301620d4e42f$exports).default,
  "hr-HR": _babelRuntimeHelpersEsmInteropRequireDefault($c91221142608664f57adc7c440d25c5c$exports).default,
  "hu-HU": _babelRuntimeHelpersEsmInteropRequireDefault($fbc3e6ba153b38828dbd4e51aedcd9c$exports).default,
  "it-IT": _babelRuntimeHelpersEsmInteropRequireDefault($e136b067ceb9d5b9a035ed9f5faa6018$exports).default,
  "ja-JP": _babelRuntimeHelpersEsmInteropRequireDefault($bc0662f1900351b0445924c5be7$exports).default,
  "ko-KR": _babelRuntimeHelpersEsmInteropRequireDefault($aae5c87ddbf07a13d7cf3cb3c706c096$exports).default,
  "lt-LT": _babelRuntimeHelpersEsmInteropRequireDefault($f9bf28c16957dd98d784eef2160689$exports).default,
  "lv-LV": _babelRuntimeHelpersEsmInteropRequireDefault($c41b94dd0457195a34a60e9d0893d89e$exports).default,
  "nb-NO": _babelRuntimeHelpersEsmInteropRequireDefault($f8ae30ecd9af5338c78bf67acc6fb392$exports).default,
  "nl-NL": _babelRuntimeHelpersEsmInteropRequireDefault($f21b1c8f7ed4d2e3b8bb19291cb8$exports).default,
  "pl-PL": _babelRuntimeHelpersEsmInteropRequireDefault($e4cc01ee8904e31a9a335f923ae8629a$exports).default,
  "pt-BR": _babelRuntimeHelpersEsmInteropRequireDefault($df8dca18e32a461efd140442fc548c72$exports).default,
  "pt-PT": _babelRuntimeHelpersEsmInteropRequireDefault($e5697b97bd7b08a87614a3a63c2233$exports).default,
  "ro-RO": _babelRuntimeHelpersEsmInteropRequireDefault($a4d94a6f59475598baa70097a6ac3345$exports).default,
  "ru-RU": _babelRuntimeHelpersEsmInteropRequireDefault($ec693ed520530f049bb03a9de8f2$exports).default,
  "sk-SK": _babelRuntimeHelpersEsmInteropRequireDefault($b620d822b146f7dbd3c5d9fd416f3640$exports).default,
  "sl-SI": _babelRuntimeHelpersEsmInteropRequireDefault($f131bcd2c7e8f9c8460b2a7928760dc$exports).default,
  "sr-SP": _babelRuntimeHelpersEsmInteropRequireDefault($eeedef4ebbff571eaf41ecd136f71$exports).default,
  "sv-SE": _babelRuntimeHelpersEsmInteropRequireDefault($db418128beb57024edf7ac9e84c764e1$exports).default,
  "tr-TR": _babelRuntimeHelpersEsmInteropRequireDefault($eb9f1cce2dfb7c8c5b9ea893e144ce0d$exports).default,
  "uk-UA": _babelRuntimeHelpersEsmInteropRequireDefault($d7aab1629c77f8c4b88e2938949da4$exports).default,
  "zh-CN": _babelRuntimeHelpersEsmInteropRequireDefault($f22a63a5549fed0b00db96652c08cf$exports).default,
  "zh-TW": _babelRuntimeHelpersEsmInteropRequireDefault($c5ea069e7a6e02a0a53f10506cd9b35$exports).default
};

const $f30f10b3f7806340277c0b2046944ca$export$MobileComboBox = /*#__PURE__*/_react.forwardRef(function MobileComboBox(props, ref) {
  props = useProviderProps(props);
  let {
    isQuiet,
    isDisabled,
    validationState,
    isReadOnly
  } = props;
  let {
    contains
  } = useFilter({
    sensitivity: 'base'
  });
  let state = useComboBoxState(_babelRuntimeHelpersEsmExtends({}, props, {
    defaultFilter: contains,
    allowsEmptyCollection: true,
    // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
    // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
    // then the closing of the tray will call setFocused(false) again due to cleanup effect)
    shouldCloseOnBlur: false
  }));
  let buttonRef = useRef();
  let domRef = useFocusableRef(ref, buttonRef);
  let {
    triggerProps,
    overlayProps
  } = useOverlayTrigger({
    type: 'listbox'
  }, state, buttonRef);
  let {
    labelProps,
    fieldProps
  } = useLabel(_babelRuntimeHelpersEsmExtends({}, props, {
    labelElementType: 'span'
  })); // Focus the button and show focus ring when clicking on the label

  labelProps.onClick = () => {
    if (!props.isDisabled) {
      buttonRef.current.focus();
      setInteractionModality('keyboard');
    }
  };

  let onClose = () => state.commit();

  return /*#__PURE__*/_react.createElement(_react.Fragment, null, /*#__PURE__*/_react.createElement(Field, _babelRuntimeHelpersEsmExtends({}, props, {
    labelProps: labelProps,
    elementType: "span",
    ref: domRef,
    includeNecessityIndicatorInAccessibilityName: true
  }), /*#__PURE__*/_react.createElement($f30f10b3f7806340277c0b2046944ca$var$ComboBoxButton, _babelRuntimeHelpersEsmExtends({}, mergeProps(triggerProps, fieldProps, {
    autoFocus: props.autoFocus
  }), {
    ref: buttonRef,
    isQuiet: isQuiet,
    isDisabled: isDisabled,
    isPlaceholder: !state.inputValue,
    validationState: validationState,
    onPress: () => !isReadOnly && state.open(null, 'manual')
  }), state.inputValue || props.placeholder || '')), /*#__PURE__*/_react.createElement(Tray, _babelRuntimeHelpersEsmExtends({
    isOpen: state.isOpen,
    onClose: onClose,
    isFixedHeight: true,
    isNonModal: true
  }, overlayProps), /*#__PURE__*/_react.createElement($f30f10b3f7806340277c0b2046944ca$var$ComboBoxTray, _babelRuntimeHelpersEsmExtends({}, props, {
    onClose: onClose,
    overlayProps: overlayProps,
    state: state
  }))));
});

var $aa1c742e7a2a16aa7981ca9e2f4a89$$interop$default = $parcel$interopDefault($aa1c742e7a2a16aa7981ca9e2f4a89$exports);
var $ed163f34af6780259e0266609fb62099$$interop$default = $parcel$interopDefault($ed163f34af6780259e0266609fb62099$exports);
var $dea09d50cdec57db7ad515930d6daf$$interop$default = $parcel$interopDefault($dea09d50cdec57db7ad515930d6daf$exports);
var $c9fb1f4220ea2c59cb2e10f0e9828c6e$$interop$default = $parcel$interopDefault($c9fb1f4220ea2c59cb2e10f0e9828c6e$exports);
var $a0b8eb4b46824fd288195a4db8814b$$interop$default = $parcel$interopDefault($a0b8eb4b46824fd288195a4db8814b$exports);

const $f30f10b3f7806340277c0b2046944ca$var$ComboBoxButton = /*#__PURE__*/_react.forwardRef(function ComboBoxButton(props, ref) {
  let {
    isQuiet,
    isDisabled,
    isPlaceholder,
    validationState,
    children,
    style,
    className
  } = props;
  let formatMessage = useMessageFormatter($f30f10b3f7806340277c0b2046944ca$var$intlMessages);
  let valueId = useId();
  let invalidId = useId();
  let validationIcon = validationState === 'invalid' ? /*#__PURE__*/_react.createElement(_spectrumIconsUiAlertMedium, {
    id: invalidId,
    "aria-label": formatMessage('invalid')
  }) : /*#__PURE__*/_react.createElement(_spectrumIconsUiCheckmarkMedium, null);

  let validation = /*#__PURE__*/_react.cloneElement(validationIcon, {
    UNSAFE_className: classNames($aa1c742e7a2a16aa7981ca9e2f4a89$$interop$default, 'spectrum-Textfield-validationIcon', classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-input-validationIcon'))
  });

  let {
    hoverProps,
    isHovered
  } = useHover({});
  let {
    buttonProps,
    isPressed
  } = useButton(_babelRuntimeHelpersEsmExtends({}, props, {
    'aria-labelledby': [props['aria-labelledby'], props['aria-label'] && !props['aria-labelledby'] ? props.id : null, valueId, validationState === 'invalid' ? invalidId : null].filter(Boolean).join(' '),
    elementType: 'div'
  }), ref);
  return /*#__PURE__*/_react.createElement(FocusRing, {
    focusClass: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'is-focused'),
    focusRingClass: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'focus-ring')
  }, /*#__PURE__*/_react.createElement("div", _babelRuntimeHelpersEsmExtends({}, mergeProps(hoverProps, buttonProps), {
    "aria-haspopup": "dialog",
    ref: ref,
    style: _babelRuntimeHelpersEsmExtends({}, style, {
      outline: 'none'
    }),
    className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup', {
      'spectrum-InputGroup--quiet': isQuiet,
      'is-disabled': isDisabled,
      'spectrum-InputGroup--invalid': validationState === 'invalid',
      'is-hovered': isHovered
    }, classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'mobile-combobox'), className)
  }), /*#__PURE__*/_react.createElement("div", {
    className: classNames($aa1c742e7a2a16aa7981ca9e2f4a89$$interop$default, 'spectrum-Textfield', {
      'spectrum-Textfield--invalid': validationState === 'invalid',
      'spectrum-Textfield--valid': validationState === 'valid',
      'spectrum-Textfield--quiet': isQuiet
    }, classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-field'))
  }, /*#__PURE__*/_react.createElement("div", {
    className: classNames($aa1c742e7a2a16aa7981ca9e2f4a89$$interop$default, 'spectrum-Textfield-input', {
      'is-hovered': isHovered,
      'is-placeholder': isPlaceholder,
      'is-disabled': isDisabled
    }, classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-input', classNames($c9fb1f4220ea2c59cb2e10f0e9828c6e$$interop$default, 'spectrum-Field-field')), classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'mobile-input'))
  }, /*#__PURE__*/_react.createElement("span", {
    id: valueId,
    className: classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'mobile-value')
  }, children)), validationState ? validation : null), /*#__PURE__*/_react.createElement("div", {
    className: classNames($a0b8eb4b46824fd288195a4db8814b$$interop$default, 'spectrum-FieldButton', {
      'spectrum-FieldButton--quiet': isQuiet,
      'is-active': isPressed,
      'is-disabled': isDisabled,
      'spectrum-FieldButton--invalid': validationState === 'invalid',
      'is-hovered': isHovered
    }, classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-FieldButton'))
  }, /*#__PURE__*/_react.createElement(_spectrumIconsUiChevronDownMedium, {
    UNSAFE_className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-Dropdown-chevron')
  }))));
});

var $f7ba17202ccc7135cfe7128957b$$interop$default = $parcel$interopDefault($f7ba17202ccc7135cfe7128957b$exports);

function $f30f10b3f7806340277c0b2046944ca$var$ComboBoxTray(props) {
  let {
    // completionMode = 'suggest',
    state,
    isDisabled,
    validationState,
    label,
    overlayProps,
    loadingState,
    onLoadMore,
    onClose
  } = props;
  let timeout = useRef(null);
  let [showLoading, setShowLoading] = useState(false);
  let inputRef = useRef();
  let buttonRef = useRef();
  let popoverRef = useRef();
  let listBoxRef = useRef();
  let layout = useListBoxLayout(state);
  let formatMessage = useMessageFormatter($f30f10b3f7806340277c0b2046944ca$var$intlMessages);
  let {
    inputProps,
    listBoxProps,
    labelProps
  } = useComboBox(_babelRuntimeHelpersEsmExtends({}, props, {
    // completionMode,
    keyboardDelegate: layout,
    buttonRef: unwrapDOMRef(buttonRef),
    popoverRef: popoverRef,
    listBoxRef,
    inputRef
  }), state);

  _react.useEffect(() => {
    focusSafely(inputRef.current); // When the tray unmounts, set state.isFocused (i.e. the tray input's focus tracker) to false.
    // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
    // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
    // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363

    return () => {
      state.setFocused(false);
    }; // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  let {
    dialogProps
  } = useDialog({
    'aria-labelledby': useId(labelProps.id)
  }, popoverRef); // Override the role of the input to "searchbox" instead of "combobox".
  // Since the listbox is always visible, the combobox role doesn't really give us anything.
  // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
  // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
  // open the virtual keyboard rather than closing the tray.

  inputProps.role = 'searchbox';
  inputProps['aria-haspopup'] = 'listbox';
  delete inputProps.onTouchEnd;

  let clearButton = /*#__PURE__*/_react.createElement(ClearButton, {
    preventFocus: true,
    "aria-label": formatMessage('clear'),
    excludeFromTabOrder: true,
    onPress: () => {
      state.setInputValue('');
      inputRef.current.focus();
    },
    UNSAFE_className: classNames($f7ba17202ccc7135cfe7128957b$$interop$default, 'spectrum-ClearButton'),
    isDisabled: isDisabled
  });

  let loadingCircle = /*#__PURE__*/_react.createElement(ProgressCircle, {
    "aria-label": formatMessage('loading'),
    size: "S",
    isIndeterminate: true,
    UNSAFE_className: classNames($f7ba17202ccc7135cfe7128957b$$interop$default, 'spectrum-Search-circleLoader', classNames($aa1c742e7a2a16aa7981ca9e2f4a89$$interop$default, 'spectrum-Textfield-circleLoader'))
  }); // Close the software keyboard on scroll to give the user a bigger area to scroll.
  // But only do this if scrolling with touch, otherwise it can cause issues with touch
  // screen readers.


  let isTouchDown = useRef(false);

  let onTouchStart = () => {
    isTouchDown.current = true;
  };

  let onTouchEnd = () => {
    isTouchDown.current = false;
  };

  let onScroll = useCallback(() => {
    if (!inputRef.current || document.activeElement !== inputRef.current || !isTouchDown.current) {
      return;
    }

    popoverRef.current.focus();
  }, [inputRef, popoverRef, isTouchDown]);
  let inputValue = inputProps.value;
  let lastInputValue = useRef(inputValue);
  useEffect(() => {
    if (loadingState === 'filtering' && !showLoading) {
      if (timeout.current === null) {
        timeout.current = setTimeout(() => {
          setShowLoading(true);
        }, 500);
      } // If user is typing, clear the timer and restart since it is a new request


      if (inputValue !== lastInputValue.current) {
        clearTimeout(timeout.current);
        timeout.current = setTimeout(() => {
          setShowLoading(true);
        }, 500);
      }
    } else if (loadingState !== 'filtering') {
      // If loading is no longer happening, clear any timers and hide the loading circle
      setShowLoading(false);
      clearTimeout(timeout.current);
      timeout.current = null;
    }

    lastInputValue.current = inputValue;
  }, [loadingState, inputValue, showLoading]);

  let onKeyDown = e => {
    // Close virtual keyboard if user hits Enter w/o any focused options
    if (e.key === 'Enter' && state.selectionManager.focusedKey == null) {
      popoverRef.current.focus();
    } else {
      inputProps.onKeyDown(e);
    }
  };

  return /*#__PURE__*/_react.createElement(FocusScope, {
    restoreFocus: true,
    contain: true
  }, /*#__PURE__*/_react.createElement("div", _babelRuntimeHelpersEsmExtends({}, mergeProps(overlayProps, dialogProps), {
    ref: popoverRef,
    className: classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'tray-dialog')
  }), /*#__PURE__*/_react.createElement(DismissButton, {
    onDismiss: onClose
  }), /*#__PURE__*/_react.createElement(TextFieldBase, {
    label: label,
    labelProps: labelProps,
    inputProps: _babelRuntimeHelpersEsmExtends({}, inputProps, {
      onKeyDown
    }),
    inputRef: inputRef,
    isDisabled: isDisabled,
    isLoading: showLoading && loadingState === 'filtering',
    loadingIndicator: loadingState != null && loadingCircle,
    validationState: validationState,
    wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly && clearButton,
    UNSAFE_className: classNames($f7ba17202ccc7135cfe7128957b$$interop$default, 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
      'spectrum-Search--invalid': validationState === 'invalid',
      'spectrum-Search--valid': validationState === 'valid'
    }, classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'tray-textfield', {
      'has-label': !!props.label
    })),
    inputClassName: classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'tray-textfield-input', classNames($f7ba17202ccc7135cfe7128957b$$interop$default, 'spectrum-Search-input')),
    validationIconClassName: classNames($f7ba17202ccc7135cfe7128957b$$interop$default, 'spectrum-Search-validationIcon')
  }), /*#__PURE__*/_react.createElement(ListBoxBase, _babelRuntimeHelpersEsmExtends({}, listBoxProps, {
    domProps: {
      onTouchStart,
      onTouchEnd
    },
    disallowEmptySelection: true,
    shouldSelectOnPressUp: true,
    focusOnPointerEnter: true,
    layout: layout,
    state: state,
    shouldUseVirtualFocus: true,
    renderEmptyState: () => loadingState !== 'loading' && /*#__PURE__*/_react.createElement("span", {
      className: classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'no-results')
    }, formatMessage('noResults')),
    UNSAFE_className: classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'tray-listbox'),
    ref: listBoxRef,
    onScroll: onScroll,
    onLoadMore: onLoadMore,
    isLoading: loadingState === 'loading' || loadingState === 'loadingMore'
  })), /*#__PURE__*/_react.createElement(DismissButton, {
    onDismiss: onClose
  })));
}

// @ts-ignore
const $a1e2444f979a947af6adfa139e4b9220$var$intlMessages = {
  "ar-AE": _babelRuntimeHelpersEsmInteropRequireDefault($aeabefb3f029f59bb5fb8536b65cc107$exports).default,
  "bg-BG": _babelRuntimeHelpersEsmInteropRequireDefault($e81215b32fa7db9a7878fa16113eb13d$exports).default,
  "cs-CZ": _babelRuntimeHelpersEsmInteropRequireDefault($bb869dea6e09984e8b04fe440dc02ec$exports).default,
  "da-DK": _babelRuntimeHelpersEsmInteropRequireDefault($eaf0e4fc84da4e97fcf8b24a0663f670$exports).default,
  "de-DE": _babelRuntimeHelpersEsmInteropRequireDefault($b56bf5d22cb0f8734a4fc5ee20a9$exports).default,
  "el-GR": _babelRuntimeHelpersEsmInteropRequireDefault($e2505be9c44ad2de362aed02038721c6$exports).default,
  "en-US": _babelRuntimeHelpersEsmInteropRequireDefault($d463e3a7667fca55cd060af86bae6dc7$exports).default,
  "es-ES": _babelRuntimeHelpersEsmInteropRequireDefault($d4e2ba5b2cf0711ecf714c846bf7985$exports).default,
  "et-EE": _babelRuntimeHelpersEsmInteropRequireDefault($e3596e15452c69f045c20524dbb2$exports).default,
  "fi-FI": _babelRuntimeHelpersEsmInteropRequireDefault($c945fdb8634b778ebd85cce64d9f$exports).default,
  "fr-FR": _babelRuntimeHelpersEsmInteropRequireDefault($ddf38499897200fd5686ac2871649df$exports).default,
  "he-IL": _babelRuntimeHelpersEsmInteropRequireDefault($a41bddea8c9d7e8d301620d4e42f$exports).default,
  "hr-HR": _babelRuntimeHelpersEsmInteropRequireDefault($c91221142608664f57adc7c440d25c5c$exports).default,
  "hu-HU": _babelRuntimeHelpersEsmInteropRequireDefault($fbc3e6ba153b38828dbd4e51aedcd9c$exports).default,
  "it-IT": _babelRuntimeHelpersEsmInteropRequireDefault($e136b067ceb9d5b9a035ed9f5faa6018$exports).default,
  "ja-JP": _babelRuntimeHelpersEsmInteropRequireDefault($bc0662f1900351b0445924c5be7$exports).default,
  "ko-KR": _babelRuntimeHelpersEsmInteropRequireDefault($aae5c87ddbf07a13d7cf3cb3c706c096$exports).default,
  "lt-LT": _babelRuntimeHelpersEsmInteropRequireDefault($f9bf28c16957dd98d784eef2160689$exports).default,
  "lv-LV": _babelRuntimeHelpersEsmInteropRequireDefault($c41b94dd0457195a34a60e9d0893d89e$exports).default,
  "nb-NO": _babelRuntimeHelpersEsmInteropRequireDefault($f8ae30ecd9af5338c78bf67acc6fb392$exports).default,
  "nl-NL": _babelRuntimeHelpersEsmInteropRequireDefault($f21b1c8f7ed4d2e3b8bb19291cb8$exports).default,
  "pl-PL": _babelRuntimeHelpersEsmInteropRequireDefault($e4cc01ee8904e31a9a335f923ae8629a$exports).default,
  "pt-BR": _babelRuntimeHelpersEsmInteropRequireDefault($df8dca18e32a461efd140442fc548c72$exports).default,
  "pt-PT": _babelRuntimeHelpersEsmInteropRequireDefault($e5697b97bd7b08a87614a3a63c2233$exports).default,
  "ro-RO": _babelRuntimeHelpersEsmInteropRequireDefault($a4d94a6f59475598baa70097a6ac3345$exports).default,
  "ru-RU": _babelRuntimeHelpersEsmInteropRequireDefault($ec693ed520530f049bb03a9de8f2$exports).default,
  "sk-SK": _babelRuntimeHelpersEsmInteropRequireDefault($b620d822b146f7dbd3c5d9fd416f3640$exports).default,
  "sl-SI": _babelRuntimeHelpersEsmInteropRequireDefault($f131bcd2c7e8f9c8460b2a7928760dc$exports).default,
  "sr-SP": _babelRuntimeHelpersEsmInteropRequireDefault($eeedef4ebbff571eaf41ecd136f71$exports).default,
  "sv-SE": _babelRuntimeHelpersEsmInteropRequireDefault($db418128beb57024edf7ac9e84c764e1$exports).default,
  "tr-TR": _babelRuntimeHelpersEsmInteropRequireDefault($eb9f1cce2dfb7c8c5b9ea893e144ce0d$exports).default,
  "uk-UA": _babelRuntimeHelpersEsmInteropRequireDefault($d7aab1629c77f8c4b88e2938949da4$exports).default,
  "zh-CN": _babelRuntimeHelpersEsmInteropRequireDefault($f22a63a5549fed0b00db96652c08cf$exports).default,
  "zh-TW": _babelRuntimeHelpersEsmInteropRequireDefault($c5ea069e7a6e02a0a53f10506cd9b35$exports).default
};

function $a1e2444f979a947af6adfa139e4b9220$var$ComboBox(props, ref) {
  props = useProviderProps(props);
  let isMobile = useIsMobileDevice();

  if (isMobile) {
    // menuTrigger=focus/manual don't apply to mobile combobox
    return /*#__PURE__*/_react.createElement($f30f10b3f7806340277c0b2046944ca$export$MobileComboBox, _babelRuntimeHelpersEsmExtends({}, props, {
      menuTrigger: "input",
      ref: ref
    }));
  } else {
    return /*#__PURE__*/_react.createElement($a1e2444f979a947af6adfa139e4b9220$var$ComboBoxBase, _babelRuntimeHelpersEsmExtends({}, props, {
      ref: ref
    }));
  }
}

const $a1e2444f979a947af6adfa139e4b9220$var$ComboBoxBase = /*#__PURE__*/_react.forwardRef(function ComboBoxBase(props, ref) {
  let {
    menuTrigger = 'input',
    shouldFlip = true,
    direction = 'bottom',
    isQuiet,
    loadingState,
    onLoadMore
  } = props;
  let formatMessage = useMessageFormatter($a1e2444f979a947af6adfa139e4b9220$var$intlMessages);
  let isAsync = loadingState != null;
  let popoverRef = useRef();
  let unwrappedPopoverRef = useUnwrapDOMRef(popoverRef);
  let buttonRef = useRef();
  let unwrappedButtonRef = useUnwrapDOMRef(buttonRef);
  let listBoxRef = useRef();
  let inputRef = useRef();
  let domRef = useFocusableRef(ref, inputRef);
  let {
    contains
  } = useFilter({
    sensitivity: 'base'
  });
  let state = useComboBoxState(_babelRuntimeHelpersEsmExtends({}, props, {
    defaultFilter: contains,
    allowsEmptyCollection: isAsync
  }));
  let layout = useListBoxLayout(state);
  let {
    buttonProps,
    inputProps,
    listBoxProps,
    labelProps
  } = useComboBox(_babelRuntimeHelpersEsmExtends({}, props, {
    keyboardDelegate: layout,
    buttonRef: unwrappedButtonRef,
    popoverRef: unwrappedPopoverRef,
    listBoxRef,
    inputRef: inputRef,
    menuTrigger
  }), state);
  let {
    overlayProps,
    placement
  } = useOverlayPosition({
    targetRef: unwrappedButtonRef,
    overlayRef: unwrappedPopoverRef,
    scrollRef: listBoxRef,
    placement: direction + " end",
    shouldFlip: shouldFlip,
    isOpen: state.isOpen,
    onClose: state.close
  }); // Measure the width of the inputfield and the button to inform the width of the menu (below).

  let [menuWidth, setMenuWidth] = useState(null);
  let {
    scale
  } = useProvider();
  let onResize = useCallback(() => {
    if (unwrappedButtonRef.current && inputRef.current) {
      let buttonWidth = unwrappedButtonRef.current.offsetWidth;
      let inputWidth = inputRef.current.offsetWidth;
      setMenuWidth(buttonWidth + inputWidth);
    }
  }, [unwrappedButtonRef, inputRef, setMenuWidth]);
  useResizeObserver({
    ref: domRef,
    onResize: onResize
  });
  useLayoutEffect(onResize, [scale, onResize]);

  let style = _babelRuntimeHelpersEsmExtends({}, overlayProps.style, {
    width: isQuiet ? null : menuWidth,
    minWidth: isQuiet ? "calc(" + menuWidth + "px + calc(2 * var(--spectrum-dropdown-quiet-offset)))" : menuWidth
  });

  return /*#__PURE__*/_react.createElement(_react.Fragment, null, /*#__PURE__*/_react.createElement(Field, _babelRuntimeHelpersEsmExtends({}, props, {
    labelProps: labelProps,
    ref: domRef
  }), /*#__PURE__*/_react.createElement($a1e2444f979a947af6adfa139e4b9220$var$ComboBoxInput, _babelRuntimeHelpersEsmExtends({}, props, {
    isOpen: state.isOpen,
    loadingState: loadingState,
    inputProps: inputProps,
    inputRef: inputRef,
    triggerProps: buttonProps,
    triggerRef: buttonRef
  }))), /*#__PURE__*/_react.createElement(Popover, {
    isOpen: state.isOpen,
    UNSAFE_style: style,
    UNSAFE_className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-popover', {
      'spectrum-InputGroup-popover--quiet': isQuiet
    }),
    ref: popoverRef,
    placement: placement,
    hideArrow: true,
    isNonModal: true,
    isDismissable: false
  }, /*#__PURE__*/_react.createElement(ListBoxBase, _babelRuntimeHelpersEsmExtends({}, listBoxProps, {
    ref: listBoxRef,
    disallowEmptySelection: true,
    autoFocus: state.focusStrategy,
    shouldSelectOnPressUp: true,
    focusOnPointerEnter: true,
    layout: layout,
    state: state,
    shouldUseVirtualFocus: true,
    isLoading: loadingState === 'loadingMore',
    onLoadMore: onLoadMore,
    renderEmptyState: () => isAsync && /*#__PURE__*/_react.createElement("span", {
      className: classNames($dea09d50cdec57db7ad515930d6daf$$interop$default, 'no-results')
    }, formatMessage('noResults'))
  })), /*#__PURE__*/_react.createElement(DismissButton, {
    onDismiss: () => state.close()
  })));
});

const $a1e2444f979a947af6adfa139e4b9220$var$ComboBoxInput = /*#__PURE__*/_react.forwardRef(function ComboBoxInput(props, ref) {
  let {
    isQuiet,
    isDisabled,
    isReadOnly,
    validationState,
    inputProps,
    inputRef,
    triggerProps,
    triggerRef,
    autoFocus,
    style,
    className,
    loadingState,
    isOpen,
    menuTrigger
  } = props;
  let {
    hoverProps,
    isHovered
  } = useHover({});
  let formatMessage = useMessageFormatter($a1e2444f979a947af6adfa139e4b9220$var$intlMessages);
  let timeout = useRef(null);
  let [showLoading, setShowLoading] = useState(false);

  let loadingCircle = /*#__PURE__*/_react.createElement(ProgressCircle, {
    "aria-label": formatMessage('loading'),
    size: "S",
    isIndeterminate: true,
    UNSAFE_className: classNames($aa1c742e7a2a16aa7981ca9e2f4a89$$interop$default, 'spectrum-Textfield-circleLoader', classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-input-circleLoader'))
  });

  let isLoading = loadingState === 'loading' || loadingState === 'filtering';
  let inputValue = inputProps.value;
  let lastInputValue = useRef(inputValue);
  useEffect(() => {
    if (isLoading && !showLoading) {
      if (timeout.current === null) {
        timeout.current = setTimeout(() => {
          setShowLoading(true);
        }, 500);
      } // If user is typing, clear the timer and restart since it is a new request


      if (inputValue !== lastInputValue.current) {
        clearTimeout(timeout.current);
        timeout.current = setTimeout(() => {
          setShowLoading(true);
        }, 500);
      }
    } else if (!isLoading) {
      // If loading is no longer happening, clear any timers and hide the loading circle
      setShowLoading(false);
      clearTimeout(timeout.current);
      timeout.current = null;
    }

    lastInputValue.current = inputValue;
  }, [isLoading, showLoading, inputValue]);
  return /*#__PURE__*/_react.createElement(FocusRing, {
    within: true,
    isTextInput: true,
    focusClass: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'is-focused'),
    focusRingClass: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'focus-ring'),
    autoFocus: autoFocus
  }, /*#__PURE__*/_react.createElement("div", _babelRuntimeHelpersEsmExtends({}, hoverProps, {
    ref: ref,
    style: style,
    className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup', {
      'spectrum-InputGroup--quiet': isQuiet,
      'is-disabled': isDisabled,
      'spectrum-InputGroup--invalid': validationState === 'invalid',
      'is-hovered': isHovered
    }, className)
  }), /*#__PURE__*/_react.createElement(TextFieldBase, {
    inputProps: inputProps,
    inputRef: inputRef,
    UNSAFE_className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-field'),
    inputClassName: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-input'),
    validationIconClassName: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-InputGroup-input-validationIcon'),
    isDisabled: isDisabled,
    isQuiet: isQuiet,
    validationState: validationState // loading circle should only be displayed if menu is open, if menuTrigger is "manual", or first time load (to stop circle from showing up when user selects an option)
    // TODO: add special case for completionMode: complete as well
    ,
    isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
    loadingIndicator: loadingState != null && loadingCircle
  }), /*#__PURE__*/_react.createElement(PressResponder, {
    preventFocusOnPress: true,
    isPressed: isOpen
  }, /*#__PURE__*/_react.createElement(FieldButton, _babelRuntimeHelpersEsmExtends({}, triggerProps, {
    ref: triggerRef,
    UNSAFE_className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-FieldButton'),
    isDisabled: isDisabled || isReadOnly,
    isQuiet: isQuiet,
    validationState: validationState
  }), /*#__PURE__*/_react.createElement(_spectrumIconsUiChevronDownMedium, {
    UNSAFE_className: classNames($ed163f34af6780259e0266609fb62099$$interop$default, 'spectrum-Dropdown-chevron')
  })))));
});
/**
 * ComboBoxes combine a text entry with a picker menu, allowing users to filter longer lists to only the selections matching a query.
 */


export const ComboBox = /*#__PURE__*/_react.forwardRef($a1e2444f979a947af6adfa139e4b9220$var$ComboBox);
